src/ansi-c/goto-conversion/goto_convert_class.h

```diff
@@ -56,6 +56,25 @@ class goto_convertt : public messaget
   std::string tmp_symbol_prefix;
   lifetimet lifetime = lifetimet::STATIC_GLOBAL;
 
+  struct clean_expr_resultt
+  {
+    std::list<irep_idt> temporaries;
```

Comment:
Are these really all temporaries created when cleaning a given `side_effect_exprt` expression or is it the absolutely minimum set that needs to be live in order to evaluate the symbol representing the cleaned `side_effect_exprt` ?
Could this set be reduced to just the one symbol representing the result of the cleaned `side_effect_exprt` ?
Lets say we are cleaning something like `foo() + bar()`
```C
  .... (instructions encoding foo())
DECL tmp1;
ASSIGN tmp1 = ;
 DEAD ... // kill all temporaries needed for foo
  .... (instructions encoding bar())
DECL tmp2;
ASSIGN tmp2 = ...;
 DEAD ... // kill all temporaries needed for bar
DECL tmp3; // final result of foo() + bar()
ASSIGN tmp3 = tmp1 + tmp2;
DEAD tmp1;
DEAD tmp2;
```
Only tmp3 needs to stay live at this point

Reply:
At some level it will indeed always be just a single one, but within recursive calls to `clean_expr` we will have multiple temporaries. Your example above demonstrates this: eventually, we will just have `tmp3`, but in between we will have `{tmp1, tmp2}` in there.